### ISSUE TITLE
private_over_fileprivate (validate_extensions) auto correction breaks code

### ISSUE THREAD

**Issue by lennartstolz** [realm/SwiftLint#2346]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Bug Report

I'm running SwiftLint v. `0.27.0`.

Running `$ swiftlint autocorrect` I face some issues that my code is broken afterwards with the following configuration:

```ruby
[...]
private_over_fileprivate:
    validate_extensions: true
[...]
```

## Before

```swift
import Foundation

class Witch { }

fileprivate extension Witch {

    fileprivate func doSomeMagic() {
        print("🧙‍♀️")
    }

}

let witch = Witch()
witch.doSomeMagic()
```

## After

```swift
import Foundation

class Witch { }

private extension Witch {

    fileprivate func doSomeMagic() {
        print("🧙‍♀️")
    }

}

let witch = Witch()
witch.doSomeMagic()
```

which doesn't run / compile anymore.

I thought rules which support auto-correction are safe to not break the code?
Or is this one of the reasons why the `validate_extensions` option is false by _default_.
Didn't find any good explanation / documentation of that configuration.

**Comment by marcelofabri** [comment]
The autocorrect should also remove the inner `fileprivate` to make it compile:

```swift
import Foundation

class Witch { }

private extension Witch {

    func doSomeMagic() {
        print("🧙‍♀️")
    }

}

let witch = Witch()
witch.doSomeMagic()
```

**Comment by marcelofabri** [comment]
@lennartstolz I think so!

The code you'd need to change is here: https://github.com/realm/SwiftLint/blob/master/Source/SwiftLintFramework/Rules/Idiomatic/PrivateOverFilePrivateRule.swift#L48-L67

The idea would be to make that `compactMap` to be a `flatMap` and when it's an extension that should be validated, it should navigate through `dictionary.substructure` and return the ranges of `fileprivate` declarations. 

Let me know if you have any questions or need some help!
